app/controllers/SessionTimeoutPageController.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.{Inject, Singleton}
+import java.time.Instant
+
+import play.api.Logger
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request, Result, Session}
+import config.{AppConfig, ErrorHandler}
+import models.errors._
+import models.ui.Text
+import services.GuidanceService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.http.logging.SessionId
+import uk.gov.hmrc.http.SessionKeys._
+import views.html.session_timeout
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@Singleton
+class SessionTimeoutPageController @Inject()(
+                                              appConfig: AppConfig,
+                                              service: GuidanceService,
+                                              errorHandler: ErrorHandler,
+                                              mcc: MessagesControllerComponents,
+                                              view: session_timeout)
+    extends FrontendController(mcc)
+    with I18nSupport {
+
+    val logger = Logger(getClass)
+
+    def getPage(processCode: String): Action[AnyContent] = Action.async { implicit request =>
+
+      implicit val messages: Messages = mcc.messagesApi.preferred(request)
+
+      val sessionId: Option[SessionId] = hc.sessionId
+
+      sessionId match {
```

Comment:
would have been useful to use the SessionFrontendController withExistingSession() here to get standard logging and use of common access to sessionId, however appreciate it would a bit of a re-write

Reply:
Think it is done this way because withExistingSession returns BadrequestError if no session and the code here is trying to distinguish between no session due to timeout or user hitting delete your answers as actions are different in each case